[distributed] Accuracy issues with test_distributed_spawn.py

### 🐛 Describe the bug

please get whls from https://github.com/intel/torch-xpu-ops/actions/runs/23637238144 or use or use gh download

```gh run download 23637238144 --repo intel/torch-xpu-ops --name Torch-XPU-Wheel-1826-23637238144-1 --dir path --pattern "*.zip"
git clone -b distributed_2.11 https://github.com/daisyden/pytorch.git
cd pytorch
pip install -r requirements.txt
pip install pytest expecttest

pytest -v ../../../../test/distributed/test_distributed_spawn.py::TestDistBackendWithSpawn::test_ddp_profiling_execution_trace
pytest -v  ../../../../test/distributed/test_distributed_spawn.py::TestDistBackendWithSpawn::test_ddp_profiling_torch_profiler ```

```
```
Process 3 exited with error code 10 and exception:
Traceback (most recent call last):
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1035, in run_test
    getattr(self, test_name)()
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 882, in wrapper
    fn()
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3560, in wrapper
    method(*args, **kwargs)
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 261, in wrapper
    return func(*args, **kwargs)
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/distributed/distributed_test.py", line 7249, in test_ddp_profiling_execution_trace
    self._validate_execution_trace_nccl(et_file)
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/distributed/distributed_test.py", line 7180, in _validate_execution_trace_nccl
    self.assertEqual(len(nccl_meta_nodes), 3)
  File "/home/pt_dist_val/miniforge3/envs/ww21_xccl_PT2.12_py3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 4578, in assertEqual
    raise error_metas.pop()[0].to_error(  # type: ignore[index]
AssertionError: Scalars are not equal!

Expected 3 but got 2.
Absolute difference: 1
Relative difference: 0.3333333333333333

To execute this test, run the following from the base repo dir:
    python test/distributed/test_distributed_spawn.py TestDistBackendWithSpawn.test_ddp_profiling_execution_trace

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```

### Versions

https://github.com/daisyden/pytorch/tree/distributed_2.11

--- a/src/xccl/ProcessGroupXCCL.cpp
+++ b/src/xccl/ProcessGroupXCCL.cpp
@@ -224,6 +224,8 @@ std::vector<at::Tensor>& TensorShelf::get() {
 }
 
 ProcessGroupXCCL::WorkXCCL::WorkXCCL(
+    std::string pgUID,
+    std::string pgDesc,
     at::Device& device,
     int rank,
     OpType opType,
@@ -234,6 +236,8 @@ ProcessGroupXCCL::WorkXCCL::WorkXCCL(
     bool enableTiming,
     bool xpuEventCacheEnabled)
     : Work(rank, opType, profilingTitle, inputs),
+      pgUID_(pgUID),
+      pgDesc_(pgDesc),
       device_(device),
       workStartTime_(std::chrono::steady_clock::now()),
       seq_(seq),
@@ -255,6 +259,8 @@ ProcessGroupXCCL::WorkXCCL::WorkXCCL(
 
 ProcessGroupXCCL::WorkXCCL::WorkXCCL(const WorkXCCL& w)
     : Work(w.rank_, w.opType_),
+      pgUID_(w.pgUID_),
+      pgDesc_(w.pgDesc_),
       device_(w.device_),
       xcclStartEvent_(w.xcclStartEvent_),
       xcclEndEvent_(w.xcclEndEvent_),
@@ -292,7 +298,7 @@ void ProcessGroupXCCL::WorkXCCL::synchronizeStream() {
 bool ProcessGroupXCCL::WorkXCCL::wait(std::chrono::milliseconds timeout) {
   RECORD_PARAM_COMMS(
       std::make_tuple(static_cast<int64_t>(this->seq_), this->isP2P_), // seq
-      std::make_tuple(pg_uid_, pg_desc_), // PG name tuple
+      std::make_tuple(pgUID_, pgDesc_), // PG name tuple
       rank_, // rank
       "wait", // collective name
       0, // inNelems
@@ -510,6 +516,8 @@ c10::intrusive_ptr<ProcessGroupXCCL::WorkXCCL> ProcessGroupXCCL::initWork(
     const std::vector<at::Tensor>& outputs,
     bool record) {
   auto r = c10::make_intrusive<ProcessGroupXCCL::WorkXCCL>(
+      pg_uid_,
+      pg_desc_,
       device,
       rank,
       opType,

--- a/src/xccl/ProcessGroupXCCL.cpp
+++ b/src/xccl/ProcessGroupXCCL.cpp
@@ -290,6 +290,20 @@ void ProcessGroupXCCL::WorkXCCL::synchronizeStream() {
 }
 
 bool ProcessGroupXCCL::WorkXCCL::wait(std::chrono::milliseconds timeout) {
+  RECORD_PARAM_COMMS(
+      std::make_tuple(static_cast<int64_t>(this->seq_), this->isP2P_), // seq
+      std::make_tuple(std::string(), std::string()), // PG name tuple
+      rank_, // rank
+      "wait", // collective name
+      0, // inNelems
+      0, // outNelems
+      at::kByte, // dType
+      std::vector<int64_t>(), // inSplitSizes
+      std::vector<int64_t>(), // outSplitSizes
+      -1,
+      -1,
+      static_cast<int>(1)); // number of device?
+
   synchronize();
 
   if (blockingWait_ || timeout != kNoTimeout) {

--- a/src/xccl/ProcessGroupXCCL.cpp
+++ b/src/xccl/ProcessGroupXCCL.cpp
@@ -292,7 +292,7 @@ void ProcessGroupXCCL::WorkXCCL::synchronizeStream() {
 bool ProcessGroupXCCL::WorkXCCL::wait(std::chrono::milliseconds timeout) {
   RECORD_PARAM_COMMS(
       std::make_tuple(static_cast<int64_t>(this->seq_), this->isP2P_), // seq
-      std::make_tuple(std::string(), std::string()), // PG name tuple
+      std::make_tuple(pg_uid_, pg_desc_), // PG name tuple
       rank_, // rank
       "wait", // collective name
       0, // inNelems

--- a/src/xccl/ProcessGroupXCCL.hpp
+++ b/src/xccl/ProcessGroupXCCL.hpp
@@ -87,6 +87,8 @@ class TORCH_API ProcessGroupXCCL : public Backend {
   class WorkXCCL : public Work {
    public:
     WorkXCCL(
+        std::string pgUID,
+        std::string pgDesc,
         at::Device& device,
         int rank,
         OpType opType,
@@ -130,6 +132,8 @@ class TORCH_API ProcessGroupXCCL : public Backend {
     }
 
    protected:
+    std::string pgUID_;
+    std::string pgDesc_;
     at::Device device_;
     std::shared_ptr<at::xpu::XPUEvent> xcclStartEvent_;
     std::shared_ptr<at::xpu::XPUEvent> xcclEndEvent_;

